### DISCUSSION TITLE
Is there some way to save values of input

### DISCUSSION THREAD

**Discussion by danylboiko95** [react-hook-form/react-hook-form#2023] [Q&A] [answered]
I'm trying to create a registration form. And I want users not to lose their data by changing the various forms, as in the example below. I read the documentation, but I found nothing. I have the idea of saving the input values via onChange in a redux form. How can I do that?
![image](https://user-images.githubusercontent.com/44903844/86351806-7f372900-bc6d-11ea-93d6-92187d556d4d.png)
![image](https://user-images.githubusercontent.com/44903844/86351855-937b2600-bc6d-11ea-87c6-2f18b95a0247.png)
![image](https://user-images.githubusercontent.com/44903844/86351895-a68df600-bc6d-11ea-802b-079ea79fb226.png)

**Comment by bluebill1049** [comment, marked as answer]
are you using V6? `shouldUnregister` for v5 you will have to use toggle visibility or store your input data in a state lib (redux).

**Reply by danylboiko95** [reply]
I'm using "react-hook-form": "^6.0.0-rc.5",

**Reply by danylboiko95** [reply]
```
const {register, handleSubmit, errors, watch} = useForm<any>(
        {
            mode: "onBlur",
            shouldUnregister: false
        }
    );
```
but the behavior the same. Data is lost

**Reply by danylboiko95** [reply]
I also lose data when I submit a form. And when I send her data, I change the form to spinner. Could this be a problem?

```
const AuthRegistration: React.FC<Props> = (props) => {
    const form = (<AuthRegistrationForm/>);
    return (
        <div>
            {props.auth.isLoading
                ? <Spinner/>
                : (<AuthTemplate
                        onClose={props.onClose}
                        title='Welcome!'
                        text='Are you ready to do something great today?'
                        form={form}/>
                )}
        </div>
    )
};
```

**Reply by bluebill1049** [reply]
AuthTeplate contain the `useForm`?

**Reply by danylboiko95** [reply]
No 

```
interface IPropsAuthTemplate {
    title: string,
    text: string,
    form: React.ReactHTMLElement<HTMLFormElement> | any,
    onClose: () => void
}

const AuthTemplate: React.FC<IPropsAuthTemplate> = (props) => {
    return (
        <div className={classes.AuthTemplate}>
            <span onClick={props.onClose} className={classes.CloseButton}>
                <FontAwesomeIcon size={"2x"} icon={faTimes}/>
            </span>
            <div className={classes.Header}>
                <h4>{props.title}</h4>
                <p>{props.text}</p>
            </div>
            <div className={classes.Form}>
                {props.form}
            </div>
        </div>
    )
};
```

**Reply by danylboiko95** [reply]
But form has 
```
const AuthRegistrationForm: React.FC<Props> = (props) => {
    const {register, handleSubmit, errors, watch} = useForm<any>(
        {
            mode: "onBlur",
            shouldUnregister: false
        }
    );
```

**Reply by bluebill1049** [reply]
it should work: https://codesandbox.io/s/shouldunregister-4e91k do you have a CSB?

**Reply by danylboiko95** [reply]
Unfortunately no. I'll try one more time with your example

**Reply by danylboiko95** [reply]
Still doesn't work :(
So the best option is to keep it in the redux store?
How can I do it when something changes? handleSubmit is not the best case

**Reply by bluebill1049** [reply]
That's really strange... well you can hoist the data into your local state then. https://codesandbox.io/s/tabs-760h9?file=/src/TabGroup.tsx

**Reply by danylboiko95** [reply]
ok, first of all, thank you for quick reply! I hope that I'll find where is the issue!

**Reply by bluebill1049** [reply]
no worries, i am keen to know what's the issue behind as well. keep me posted.

**Reply by danylboiko95** [reply]
Yeah, sure!

**Reply by danylboiko95** [reply]
Also, when I toggle modal data is saved. But when I close and open my auth modal, my data is lost. I suppose there's a problem with the parenting elements. I have to do something about them

![image](https://user-images.githubusercontent.com/44903844/86357357-63845080-bc76-11ea-9de7-c5a96e53dba8.png)

**Reply by danylboiko95** [reply]
I think that the best way is will be to save data into redux because I want to keep it even when I change the path of the URL
